code/workspaces/auth-service/src/dataaccess/userRolesRepository.js

```diff
@@ -56,12 +56,23 @@ async function getUsers() {
   return Object.values(usersMap); // return users
 }
 
+function combineRoles(roles1, roles2) {
+  return {
+    ...roles1,
+    ...roles2,
+    projectRoles: [
+      ...(roles1 ? roles1.projectRoles : []),
+      ...(roles2 ? roles2.projectRoles : []),
+    ],
+  };
+}
+
 async function getAllUsersAndRoles() {
   const allRoles = await UserRoles().find().exec();
   const usersMap = allRoles
     .filter(roles => roles.userName) // only take users with known user names
     .reduce((uniqueUsersMap, roles) => { // convert to map, keyed by userId
-      uniqueUsersMap[roles.userId] = addDefaults(roles); // eslint-disable-line no-param-reassign
+      uniqueUsersMap[roles.userId] = combineRoles(uniqueUsersMap[roles.userId], addDefaults(roles)); // eslint-disable-line no-param-reassign
```

Comment:
Am I right in thinking this is to handle the case where a user is defined more than once? If that is the case, I don't think it is immediately obvious from the code so it would be good to have a comment or alternative method that makes this clear.

Reply:
Comment added